src/signing_key.rs

```diff
@@ -106,6 +115,68 @@ impl From<[u8; 32]> for SigningKey {
     }
 }
 
+impl<'a> TryFrom<PrivateKeyInfo<'a>> for SigningKey {
+    type Error = Error;
+    fn try_from(pki: PrivateKeyInfo) -> Result<Self, Error> {
+        if pki.algorithm == ALGORITHM_ID {
+            SigningKey::try_from(pki.private_key)
+        } else {
+            Err(Error::MalformedSecretKey)
+        }
+    }
+}
+
+impl ToPrivateKey for SigningKey {
+    fn to_pkcs8_der(&self) -> Result<PrivateKeyDocument, pkcs8::Error> {
+        // In RFC 8410, the octet string containing the private key is encapsulated by
+        // another octet string. Just add octet string bytes to the key.
+        let octetstring_bytes_string = "0420";
+        let mut octetstring_array = [0u8; 2];
+        hex::decode_to_slice(octetstring_bytes_string, &mut octetstring_array as &mut [u8]).ok();
+
+        let mut final_key = [0 as u8; 34];
+        let mut key_byte_sequence =  octetstring_array.iter()
+            .chain(self.seed.iter());
+        let _bytes_written = key_byte_sequence
+            .by_ref()
+            .zip(final_key.as_mut())
+            .fold(0, | cnt, (item, slot) | {
+                *slot = item.clone(); cnt+1
+            });
+
+        Ok(PrivateKeyInfo::new(ALGORITHM_ID, &final_key).into())
+    }
+}
+
+impl FromPrivateKey for SigningKey {
+    fn from_pkcs8_private_key_info(pki: PrivateKeyInfo<'_>) -> Result<Self, pkcs8::Error> {
+        // Split off the extra octet string bytes.
+        let (octetstring_prefix, private_key) = pki.private_key.split_at(2);
+        if hex::encode(octetstring_prefix).ne("0420") {
+            Err(pkcs8::Error::Asn1(der::Error::new(der::ErrorKind::Failed, der::Length::ZERO)))
+        }
+        else {
+            SigningKey::try_from(private_key).map_err(|_| pkcs8::Error::Crypto)
+        }
```

Comment:
Not quite sure this is valid, but I'd suggest something like this:

```suggestion
        match &pki.private_key {
            [0x04, 0x20, private_key @ ..] => SigningKey::try_from(private_key).map_err(|_| pkcs8::Error::Crypto),
            _ => Err(der::Tag::OctetString.value_error().into())
        }
```